target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom

```diff
@@ -9,45 +9,29 @@ board=$(board_name)
 case "$FIRMWARE" in
 "ath9k-eeprom-ahb-18100000.wmac.bin")
 	case $board in
-	aruba,ap-115)
-		caldata_extract "oemdata" 0x1000 0x440
-		;;
-	asus,pl-ac56|\
-	asus,rp-ac51|\
-	asus,rp-ac66)
-		caldata_extract "art" 0x1000 0x440
-		;;
 	avm,fritz1750e|\
 	avm,fritz4020|\
 	avm,fritz450e|\
 	avm,fritzdvbc)
 		caldata_extract_reverse "urlader" 0x1541 0x440
 		;;
-	dlink,dir-505)
-		caldata_extract "art" 0x1000 0x440
-		ath9k_patch_mac $(mtd_get_mac_text "mac" 0x4)
-		;;
 	dlink,dir-842-c1|\
 	dlink,dir-842-c2|\
 	dlink,dir-842-c3|\
 	nec,wf1200cr|\
 	nec,wg1200cr|\
 	wd,mynet-n600|\
 	wd,mynet-n750)
-		caldata_extract "art" 0x1000 0x440
 		ath9k_patch_mac $(mtd_get_mac_ascii devdata "wlan24mac")
```

Comment:
Please create another commit to port these `ath9k_patch_mac` to `target/linux/ath79/{generic,nand,tiny}/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac` scripts. They won't take effect here.

Reply:
I was under the impression mtd_get_mac_ascii wasn't portable to nvmem.

Reply:
u-boot-env nvmem can only handle variable `ethaddr`. So we need to set these mac address via `10_fix_wifi_mac` scripts. Please take a look on https://github.com/openwrt/openwrt/commit/7e71eef5edbbe3670621cfdacb3731e1bd9a8ec0.

If device has ath10k pcie NIC, the ath9k NIC is `phy1`, otherwise it's `phy0`.

Reply:
If the ASCII MAC address is located at a fixed address, it can be read this way: https://github.com/openwrt/openwrt/blob/main/target/linux/ath79/dts/ar9342_zyxel_nwa11xx.dtsi#L95-L111
But I guess that's not the point here...

Reply:
That's mtd_get_mac_text

Reply:
So far I got

```
diff --git a/target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom b/target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom
index 1b9932ec10..c1050dbc02 100644
--- a/target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom
+++ b/target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom
@@ -52,16 +52,6 @@ case "$FIRMWARE" in
 	avm,fritz300e)
 		caldata_extract_reverse "urloader" 0x1541 0x440
 		;;
-	enterasys,ws-ap3705i)
-		ath9k_patch_mac $(mtd_get_mac_ascii u-boot-env0 RADIOADDR0)
-		;;
-	wd,mynet-n600|\
-	wd,mynet-n750)
-		ath9k_patch_mac $(mtd_get_mac_ascii devdata "wlan5mac")
-		;;
-	wd,mynet-wifi-rangeextender)
-		ath9k_patch_mac $(nvram get wl0_hwaddr)
-		;;
 	*)
 		caldata_die "board $board is not supported yet"
 		;;
diff --git a/target/linux/ath79/generic/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac b/target/linux/ath79/generic/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac
index 1af3f52dbe..760eba0f68 100644
--- a/target/linux/ath79/generic/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac
+++ b/target/linux/ath79/generic/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac
@@ -67,6 +67,10 @@ case "$board" in
 		[ "$PHYNBR" -eq 0 ] && \
 		macaddr_add $(cat /sys/class/net/eth0/address) 1 > /sys${DEVPATH}/macaddress
 		;;
+	enterasys,ws-ap3705i)
+		[ "$PHYNBR" -eq 0 ] && \
+		mtd_get_mac_ascii u-boot-env0 RADIOADDR0 > /sys${DEVPATH}/macaddress
+		;;
 	extreme-networks,ws-ap3805i)
 		[ "$PHYNBR" -eq 0 ] && \
 		mtd_get_mac_ascii cfg1 RADIOADDR0 > /sys${DEVPATH}/macaddress
@@ -90,7 +94,9 @@ case "$board" in
 		;;
 	nec,wf1200cr|\
 	nec,wg1200cr|\
-	qihoo,c301)
+	qihoo,c301|\
+	wd,mynet-n600|\
+	wd,mynet-n750)
 		[ "$PHYNBR" -eq 0 ] && \
 			mtd_get_mac_ascii devdata wlan5mac > /sys${DEVPATH}/macaddress
 		;;
@@ -109,6 +115,10 @@ case "$board" in
 		[ "$PHYNBR" -eq 0 ] && \
 			macaddr_add $base_mac 2 > /sys${DEVPATH}/macaddress
 
+		;;
+	wd,mynet-wifi-rangeextender)
+		[ "$PHYNBR" -eq 0 ] && \
+		nvram get wl0_hwaddr > /sys${DEVPATH}/macaddress
 		;;
 	zyxel,nwa1123-ac)
 		[ "$PHYNBR" -eq 0 ] && \

```

not sure if right.

Reply:
This was done.